OneCommunityGlobal/HighestGoodNetworkApp
# Xiao F create the ability for owners to approve and apply suggested changes on a task

# Description
![CleanShot 2023-10-25 at 20 31 30](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/3d06e02c-c587-4827-a1b8-dce227c9fb98)

## Related PRS (if any):
[PR#949](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/949)

## Main changes explained:
- Update file `src/components/Header/Header.jsx` for fetching notifications for owners

## How to test:
1. check into current branch
2. Start the FE by `npm i` and `npm run start:local`
3. log in as a volunteer user
4. go to dashboard -> Tasks -> click on task name -> suggest -> make some changes to the task -> update
5. log in as owner user
6. go to dashboard -> left side of 'Dashboard' should show the number suggested changes -> click on the number -> view suggestion -> approve -> changes should reflect on the task

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/7dd5e58d-e0b8-438a-82e4-d97ef2f8cc6b



## Note:
1. For BE, this `TypeError: Cannot read property 'permissions' of null` error exists on dev branch:
![CleanShot 2023-10-25 at 22 53 07](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/8146475d-4be9-40db-ad19-8064c2f4a623)

2. Manager has both edit and suggest permissions in the database, but edit permission overrides suggest so that 'suggest' button won't appear for manager.
